optimizer/typecomputer.go

```diff
@@ -13,20 +13,18 @@
 
 package optimizer
 
-import (
-	"github.com/pingcap/tidb/ast"
-	"github.com/pingcap/tidb/stmt"
-)
+import "github.com/pingcap/tidb/ast"
 
-// Compile compiles a ast.Node into a executable statement.
-func Compile(node ast.Node) (stmt.Statement, error) {
-	switch v := node.(type) {
-	case *ast.SetStmt:
-		return compileSet(v)
-	}
-	return nil, nil
+// typeComputer is an ast Visitor that
+// Compute types for ast.ExprNode.
```

Comment:
The type of an expression is computed by its operands and operator.
the type of ( "1" + 1) is double.